### PR TITLE
TASK: Add support for PostgreSQL

### PR DESCRIPTION
**Pull request by dfeyer**

### CODE REVIEW

File: Migrations/Postgresql/Version20170718190551.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Neos\Flow\Persistence\Doctrine\Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs! This block will be used as the migration description if getDescription() is not used.
```
Comment (kdambekalns) [inline Migrations/Postgresql/Version20170718190551.php]: Get rid of this.

File: Migrations/Postgresql/Version20170718190551.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ */
+class Version20170718190551 extends AbstractMigration
+{
+
+    /**
+     * @return string
+     */
+    public function getDescription()
+    {
+        return '';
```
Comment (kdambekalns) [inline Migrations/Postgresql/Version20170718190551.php]: … but fill this.

File: Migrations/Postgresql/Version20170718190551.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        $this->addSql('CREATE TABLE ttree_contentrepositoryimporter_domain_model_recordmapping (persistence_object_identifier VARCHAR(40) NOT NULL, creationdate TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL, modificationdate TIMESTAMP(0) WITHOUT TIME ZONE DEFAULT NULL, importerclassname VARCHAR(255) NOT NULL, importerclassnamehash VARCHAR(32) NOT NULL, externalidentifier VARCHAR(255) NOT NULL, externalrelativeuri VARCHAR(255) DEFAULT NULL, nodeidentifier VARCHAR(255) NOT NULL, nodepath VARCHAR(4000) NOT NULL, nodepathhash VARCHAR(32) NOT NULL, PRIMARY KEY(persistence_object_identifier))');
+        $this->addSql('CREATE INDEX nodepathhash ON ttree_contentrepositoryimporter_domain_model_recordmapping (nodepathhash)');
+        $this->addSql('CREATE INDEX nodeidentifier ON ttree_contentrepositoryimporter_domain_model_recordmapping (nodeidentifier)');
+        $this->addSql('CREATE UNIQUE INDEX UNIQ_8C1932AD8FC282F11EE98D63 ON ttree_contentrepositoryimporter_domain_model_recordmapping (importerclassnamehash, externalidentifier)');
+        $this->addSql('ALTER TABLE neos_neos_eventlog_domain_model_event ADD externalidentifier VARCHAR(255) DEFAULT NULL');
```
Comment (kdambekalns) [inline Migrations/Postgresql/Version20170718190551.php:34]: This seems unrelated. If it isn't it should still not happen, that is another package!

File: Migrations/Postgresql/Version20170718190551.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+     * @param Schema $schema
+     * @return void
+     */
+    public function down(Schema $schema)
+    {
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'postgresql', 'Migration can only be executed safely on "postgresql".');
+        
+        $this->addSql('DROP TABLE ttree_contentrepositoryimporter_domain_model_import');
+        $this->addSql('DROP TABLE ttree_contentrepositoryimporter_domain_model_recordmapping');
+        $this->addSql('ALTER TABLE neos_neos_eventlog_domain_model_event DROP externalidentifier');
```
Comment (kdambekalns) [inline Migrations/Postgresql/Version20170718190551.php:47]: This and the following two lines seem unrelated. If it isn't it should still not happen, that is another package!

**Review by kdambekalns — CHANGES_REQUESTED** [review]

File: Migrations/Postgresql/Version20170718190551.php (hunk 3 above)
Comment (dfeyer) [reply Migrations/Postgresql/Version20170718190551.php:34]: @kdambekalns  I extend the NodeEvent to use register action from the importer and add one column, in Ttree\ContentRepositoryImporter\Domain\Model\Event

Comment (dfeyer) [reply Migrations/Postgresql/Version20170718190551.php:34]: Basically I have the same thing in the MySQL migration. Maybe in a future version I should use my own event log.

Comment (kdambekalns) [reply Migrations/Postgresql/Version20170718190551.php:34]: Uhm, well. Yes, probably.

**Review by kdambekalns — APPROVED** [review]
Well, fine with me. I think the adjustment to the events table is weird, but hey. Who am I to judge. :D
